Sources/Sentry/SentrySessionTracker.m

```diff
@@ -84,17 +85,20 @@ - (void)stop
 
 - (void)didBecomeActive
 {
-    NSDate *sessionEnded
-        = nil == self.lastInForeground ? [self.currentDateProvider date] : self.lastInForeground;
-    NSTimeInterval secondsInBackground =
-        [[self.currentDateProvider date] timeIntervalSinceDate:sessionEnded];
     SentryHub *hub = [SentrySDK currentHub];
-    if (secondsInBackground * 1000 > (double)(self.options.sessionTrackingIntervalMillis)) {
-        [hub endSessionWithTimestamp:sessionEnded];
+    self.lastInForeground = [[[hub getClient] fileManager] readTimestampLastInForeground];
+
+    if (nil == self.lastInForeground) {
         [hub startSession];
+    } else {
+        NSTimeInterval secondsInBackground =
+            [[self.currentDateProvider date] timeIntervalSinceDate:self.lastInForeground];
+
+        if (secondsInBackground * 1000 >= (double)(self.options.sessionTrackingIntervalMillis)) {
+            [hub endSessionWithTimestamp:self.lastInForeground];
```

Comment:
M: I'm wondering if it could be the case that theres no session to be ended, how `endSessionWithTimestamp` will behave? no exception or anything?

Reply:
The hub doesn't do anything, see https://github.com/getsentry/sentry-cocoa/blob/fix/abnormal-sessions/Sources/Sentry/SentryHub.m#L86-L89. I'm going to add a test for this.